docs/proposals/tool-to-backfill-tsdb-blocks.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "Tools to backfill TSDB blocks"
+linkTitle:
+weight: 1
+slug:
+---
+
+- Author: [Siddharth Asthana](https://github.com/edith007)
+- Date: March 2023
+- Status: draft
+
+## Overview
+
+There might be a lot of cases where the users might already be using Prometheus and later decides to switch to Cortex. Cortex will take care of the metrics which will be generated after the switch, but at present there is no way to backfill the old Prometheus TSDB blocks to Cortex. This proposal is to add the ability in cortex to fetch the Prometheus TSDB blocks and backfill the data.
+
+## Detailed Solution
+
+We can create a new service called `Importer` which will be responsible for bringing in the old Prometheus TSDB blocks to Cortex.
+
+### How Importer will work
+
+The importer service will have one API endpoints: `import`.
+
+`import`:
+
+- It will take a link to where the old Prometheus TSDB blocks are stored
+- The API will return an ID. This ID can be used to get the status of the import operation
+- Cortex will download this block from the link received and upload to the long term storage
+
+### How will Cortex import the old TSDB blocks
+
+Cortex will import the old TSDB blocks by using the `cortex-cli` tool, which will be used by
+the user to initiate the import process. The `cortex-cli` will take care of the process of
+uploading the blocks to cortex. The blocks can be store in either local storage or remote
+storage (such as GCS or S3), depending on the user's choice.
+
+When the user runs the `cortex-cli` import command, it will initiate the import process. The
+CLI will read the block from the specified location and upload them to the Cortex cluster. The
+`cortex-cli` will also provide feedback to the user, such as progress bars and status updates.
```

Comment:
I'm wondering how the API and the cortex CLI interact together. Given the ID returned from the original API upload call, how would the status of that ID be updated?